### PR TITLE
STOMP refactor in accordance with MQTT (shared connection using config node) + client ACK

### PR DESCRIPTION
**Pull request by OlivierVerhaegen**
## Types of changes
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes
I did a refactor of the STOMP nodes to be in accordance with the default Node-Red MQTT nodes as the connection is handled in a better way (shared for all nodes using the same config node). In the current version of the STOMP nodes each node creates its own connection which causes unneeded overhead (and confusion when managing connections on the server).  

Additionally a node for sending client Acknowledgments was added. This enables us to keep messages in the queue until the handling was successful on the client side.

Partly fix: https://github.com/node-red/node-red-nodes/issues/143
NACK is currently not implemented as I don't really see a use case for it (I don't find a detailed explanation about it, from my findings the server controls what happend on NACK)

Is it plausible to make this a V1.0.0 release for the STOMP node or when will it reach V1?

## Checklist
- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by dceejay** [comment]
Thanks for this. On vacation this week - will try to look at it next week.

**Comment by OlivierVerhaegen** [comment]
@dceejay Hi, I'm sorry to push like this but we would like to implement this change without pointing to my own fork. Any news about implementing this PR. Thanks a lot!

File: io/stomp/18-stomp.html [hunk 1]
```diff
@@ -146,10 +159,10 @@
         defaults: {
             server: {required:true},
             port: {value:61613,required:true,validate:RED.validators.number()},
-            protocolversion: {value:"1.0",required:true},
+            protocolVersion: {value:"1.0",required:true},
             vhost: {},
-            reconnectretries: {value:"0",required:true,validate:RED.validators.number()},
-            reconnectdelay: {value:"0.5",required:true,validate:RED.validators.number()},
+            reconnectRetries: {value:0,required:true,validate:RED.validators.number()},
+            reconnectDelay: {value:1,required:true,validate:RED.validators.number()},
```
[the remark this answers is not in the document]
Comment (hardillb) [reply io/stomp/18-stomp.html:165]: Just to be clear, the rename means that any old values will be ignored and the defaults will be used in the case of an upgrade. Also the old values with the old names will be left in the flow JSON going forward unnecessarily bloating the flow.

While yes camel case would be nice, it's not worth breaking existing flows, even with a jump to v1.0.0

**Comment by dceejay** [comment]
Thanks for your efforts on this.

**Pull request merged**